### ISSUE TITLE
[BUG] fleet-agent uninstall the Harvester with a failed upgrade

### ISSUE THREAD

**Issue by guangbochen** [harvester/harvester#2107]
**Describe the bug**
The `fleet-agent` will delete the Harvester chart when the upgrade is failed, more details are attached to the `fleet-agent` log.

**To Reproduce**
- still investigating how to reproduce it stably as it only happens once in our testing env.

**Expected behavior**
The `fleet-agent` should never uninstall the Harvester chart.

**Support bundle**
[fleet-agent-989784c4-d7j5n.log](https://github.com/harvester/harvester/files/8395612/fleet-agent-989784c4-d7j5n.log)
[fleet-agent-57497ff7dc-6d4mk.log](https://github.com/harvester/harvester/files/8395611/fleet-agent-57497ff7dc-6d4mk.log)

**Environment:**
 - Harvester ISO version:  v1.0.1-rc2
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): Bare-mental server with 4 nodes.

**Comment by FrankYang0529** [comment]
I found that if we delete releases secrets like `harvester-system/sh.helm.release.v1.harvester.v1`, then related cattle App `harvester-system/harvester` will be deleted too.

However, my fleet log is a little different from Guangbo's. My fleet will try to install `harvester` again, not uninstall.

```
level=info msg="getting history for release harvester"
level=info msg="Helm: Installing mcc-harvester"
level=info msg="CRD kubevirts.kubevirt.io is already present. Skipping."
level=info msg="CRD volumesnapshotclasses.snapshot.storage.k8s.io is already present. Skipping."
level=info msg="CRD volumesnapshotcontents.snapshot.storage.k8s.io is already present. Skipping."
level=info msg="CRD volumesnapshots.snapshot.storage.k8s.io is already present. Skipping."
level=info msg="getting history for release harvester"
level=info msg="getting history for release harvester"
level=error msg="error syncing 'cluster-fleet-local-local-1a3d67d0a899/mcc-harvester': handler bundle-deploy: rendered manifests contain a resource that already exists. Unable to continue with install: ConfigMap \"longhorn-storageclass\" in namespace \"longhorn-system\" exists and cannot be imported into the current release: invalid ownership metadata; annotation validation error: missing key \"meta.helm.sh/release-name\": must be set to \"harvester\"; annotation validation error: missing key \"meta.helm.sh/release-namespace\": must be set to \"harvester-system\", requeuing"
```

**Comment by FrankYang0529** [comment]
Deleting any one of `harvester` ManagedChart, `mcc-harvester` Bundle and `mcc-harvester` BundleDeployment will trigger all harvester-related resources to be deleted. Harvester system can be recovered if users only delete `mcc-harvester` Bundle or `macc-harvester` BundleDepolyment, but we will lose data like `VM`.

I think we should add a webhook to prevent users from removing any of `harvester` ManagedChart, `mcc-harvester` Bundle or `mcc-harvester` BundleDeployment.

**Comment by TachunLin** [comment]
Verified **fixed** on `master-e2985453-head` (04/28). Close this issue.

### Result
The following ManagedCharts, Bundles and BundleDeployments have been protected, can't delete them through kubectl command.

1. Unable to delete `harvester` and `harvester-crd` ManagedChart,

  ```
  harvester-220428:~ # kubectl get ManagedChart -A
  NAMESPACE     NAME                                      AGE
  fleet-local   harvester                                 29m
  fleet-local   harvester-crd                             29m
  fleet-local   local-managed-system-upgrade-controller   29m
  fleet-local   rancher-monitoring                        29m
  fleet-local   rancher-monitoring-crd                    29m
  ```

  ```
  harvester-220428:~ # kubectl delete ManagedChart harvester -n fleet-local
  The request is invalid: : Delete managedchart fleet-local/harvester is prohibited
  harvester-220428:~ # kubectl delete ManagedChart harvester-crd -n fleet-local
  The request is invalid: : Delete managedchart fleet-local/harvester-crd is prohibited
  ```

2. Unable to delete `mcc-harvester` and `mcc-harvester-crd` Bundle

  ```
  harvester-220428:~ # kubectl get Bundle -A
  NAMESPACE     NAME                                          BUNDLEDEPLOYMENTS-READY   STATUS
  fleet-local   fleet-agent-local                             1/1                       
  fleet-local   local-managed-system-agent                    1/1                       
  fleet-local   mcc-harvester                                 0/1                       Modified(1) [Cluster fleet-local/local]; configmap.v1 longhorn-system/longhorn-storageclass missing
  fleet-local   mcc-harvester-crd                             1/1                       
  fleet-local   mcc-local-managed-system-upgrade-controller   1/1                       
  fleet-local   mcc-rancher-monitoring                        1/1                       
  fleet-local   mcc-rancher-monitoring-crd                    1/1
  ```

  ```
  harvester-220428:~ # kubectl delete Bundle mcc-harvester -n fleet-local
  The request is invalid: : Delete bundle fleet-local/mcc-harvester is prohibited
  harvester-220428:~ # kubectl delete Bundle mcc-harvester-crd -n fleet-local
  The request is invalid: : Delete bundle fleet-local/mcc-harvester-crd is prohibited
  ```

3. Unable to delete `mcc-harvester` and `mcc-harvester-crd` BundleDeployment

  ```
  harvester-220428:~ # kubectl get BundleDeployment -A
  NAMESPACE                                NAME                                          DEPLOYED   MONITORED   STATUS
  cluster-fleet-local-local-1a3d67d0a899   fleet-agent-local                             True       True        
  cluster-fleet-local-local-1a3d67d0a899   local-managed-system-agent                    True       True        
  cluster-fleet-local-local-1a3d67d0a899   mcc-harvester                                 True       True        configmap.v1 longhorn-system/longhorn-storageclass missing
  cluster-fleet-local-local-1a3d67d0a899   mcc-harvester-crd                             True       True        
  cluster-fleet-local-local-1a3d67d0a899   mcc-local-managed-system-upgrade-controller   True       True        
  cluster-fleet-local-local-1a3d67d0a899   mcc-rancher-monitoring                        True       True        
  cluster-fleet-local-local-1a3d67d0a899   mcc-rancher-monitoring-crd                    True       True
  ```

  ```
  harvester-220428:~ # kubectl delete BundleDeployment mcc-harvester -n cluster-fleet-local-local-1a3d67d0a899
  The request is invalid: : Delete bundledeployment cluster-fleet-local-local-1a3d67d0a899/mcc-harvester is prohibited
  harvester-220428:~ # kubectl delete BundleDeployment mcc-harvester-crd -n cluster-fleet-local-local-1a3d67d0a899
  The request is invalid: : Delete bundledeployment cluster-fleet-local-local-1a3d67d0a899/mcc-harvester-crd is prohibited
  ```


### Test Information
* Test Environment: 1 node harvester on local kvm machine
* Harvester version: master-e2985453-head (04/28)

### Verify Steps
1. ssh to harvester VIP node
2. Run `sudo -i`
3. Delete the `harvester` and `harvester-crd` ManagedChart,
4. Delete the `mcc-harvester` and `mcc-harvester-crd` Bundle
5. Delete the `mcc-harvester` and `mcc-harvester-crd` BundleDeployment
